devel/githubContrib.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// go run githubContrib.go to update Contributions.txt
+
+// This script goes from org -> repos (skipping forks) -> contributors -> user
+// -> guess/normalize the company and count contribs
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+)
+
+// Check checks for non nil error and dies upon error.
+func checkOrDie(err error, msg string) {
+	if err != nil {
+		log.Fatal(msg, err)
+	}
+}
+
+// tokenFromEnv gets auth token from the env.
+func tokenFromEnv() string {
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		log.Fatal("Need to have GITHUB_TOKEN set in the env")
+	}
+	return token
+}
+
+// GitHubAPIURL returns the full v3 rest api for a given path.
+func GitHubAPIURL(path string) string {
```

Comment:
I explained why in previous set of comments but ok, will unexport everything if that helps